src/material-experimental/mdc-progress-spinner/progress-spinner.ts

```diff
@@ -0,0 +1,268 @@
+import {ANIMATION_MODULE_TYPE} from '@angular/platform-browser/animations';
+import {
+  MAT_PROGRESS_SPINNER_DEFAULT_OPTIONS,
+  MatProgressSpinnerDefaultOptions
+} from '@angular/material/progress-spinner';
+import {coerceNumberProperty} from '@angular/cdk/coercion';
+
+// Boilerplate for applying mixins to MatProgressBar.
+/** @docs-private */
+class MatProgressSpinnerBase {
+  constructor(public _elementRef: ElementRef) {
+  }
+}
+
+const _MatProgressSpinnerMixinBase: CanColorCtor & typeof MatProgressSpinnerBase =
+  mixinColor(MatProgressSpinnerBase, 'primary');
+
+/** Possible mode for a progress spinner. */
+export type ProgressSpinnerMode = 'determinate' | 'indeterminate';
+
+/**
+ * Base reference size of the spinner.
+ * @docs-private
+ */
+const BASE_SIZE = 100;
+
+/**
+ * Base reference stroke width of the spinner.
+ * @docs-private
+ */
+const BASE_STROKE_WIDTH = 10;
+
+@Component({
+  selector: 'mat-progress-spinner',
+  exportAs: 'matProgressSpinner',
+  host: {
+    'role': 'progressbar',
+    'class': 'mat-mdc-progress-spinner',
+    '[class._mat-animation-noopable]': `_noopAnimations`,
+    '[style.width.px]': 'diameter',
+    '[style.height.px]': 'diameter',
+    '[attr.aria-valuemin]': 'mode === "determinate" ? 0 : null',
+    '[attr.aria-valuemax]': 'mode === "determinate" ? 100 : null',
+    '[attr.aria-valuenow]': 'mode === "determinate" ? value : null',
+    '[attr.mode]': 'mode',
+  },
+  inputs: ['color'],
+  templateUrl: 'progress-spinner.html',
+  styleUrls: ['progress-spinner.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MatProgressSpinner extends _MatProgressSpinnerMixinBase implements AfterViewInit,
+  OnDestroy,
+  CanColor {
+  /** Whether the _mat-animation-noopable class should be applied, disabling animations.  */
+  _noopAnimations: boolean;
+  /** Implements all of the logic of the MDC circular progress. */
+  private _foundation: MDCCircularProgressFoundation | undefined;
+  private _rootElement: HTMLElement;
+  private _determinateCircle: HTMLElement;
+  /** Adapter used by MDC to interact with the DOM. */
+  private _adapter: MDCCircularProgressAdapter = {
+    addClass: (className: string) => this._rootElement.classList.add(className),
+    hasClass: (className: string) => this._rootElement.classList.contains(className),
+    removeClass: (className: string) => this._rootElement.classList.remove(className),
+    removeAttribute: (name: string) => this._rootElement.removeAttribute(name),
+    setAttribute: (name: string, value: string) => this._rootElement.setAttribute(name, value),
+    getDeterminateCircleAttribute: (attributeName: string) => this._determinateCircle.getAttribute(
+      attributeName),
+    setDeterminateCircleAttribute: (attributeName: string,
+                                    value: string) => this._determinateCircle.setAttribute(
+      attributeName,
```

Comment:
The formatting here seems odd. If it doesn't fit on a single line, you can break the arrow function into multiple lines:

```
fn: () => {
  // do stuff
}
```